### ISSUE TITLE
RVM installs with default ruby but fails to create gemset

### ISSUE THREAD

**Issue by gregmark** [maestrodev/puppet-rvm#89]
I am running maestrodev-rvm (v1.11.0) with Puppet 3.7.5. The client is running CentOS 6.4. 

I've gotten early versions to work by running puppet twice but lately I have been encountering a new failure after the RVM module is run and I try to create gemsets with another module (I was having the same problem with v1.8.2). I have tried running only the rvm module and then adding the gemset module (gems for use with Capistrano), but I end up with this every time:

http://pastebin.com/zPAcJ5vm

I have not changed anything within the module itself.  Parameter data is stored in hiera:

http://pastebin.com/egNXwfTN

Finally here is the sole .pp file from the "cap" module I created to install the gemset:

http://pastebin.com/RPeDSUxB

I understand that evaluation order of the catalog can cause issues, but this seems unrelated. Can you shed any light as to what's happening here?

Thanks
--Greg Chavez

**Comment by carlossg** [comment]
some questions

* the log is from the first puppet run? or following ones?
* is the ruby actually installed? output of `rvm list strings`

Check the debug log from puppet. It could be that the commands executed to list installed rubies are parsed incorrectly and Puppet thinks the ruby is already installed so it doesn't try to install and that's why it fails later. 
The debug log should say what commands are run and what's the output

**Comment by gregmark** [comment]
Well, if I try to do it all on one run - the rvm module runs, then the capistrano gemset module with the rvm require, I get the output you see. If I make a run without the cap gemset module, the rvm module completes, but the a successive run that includes the cap gemset module fails with the same log errors.

[user@node ~]$ rvm list strings
ruby-1.9.3-p484

I also ran the debugs and it shows that the ruby istalled properly. This is verified with rvm rubies list.

Then I thought of an offhand comment one of my co-workers said about the version of rvm requiring this that or the other. So I ran rvm implode, edited hiera to change rvm::version: from 1.25.15 to 1.26.11.....

And it worked in one run. Is there a warning for this in this module's instructions? I'll have to look. Thanks for the suggestions.
